### PR TITLE
git-lfs: udpate to 3.3.0

### PR DESCRIPTION
**Pull request by pascal-huber**
#### Testing the changes
- I tested the changes in this PR: **briefly**

#### Local build testing
- I built this PR locally for my native architecture, x86_64-glibc
- I built this PR locally for these architectures:
  - x86_64-musl (crossbuild)
  - aarch64 (crossbuild)
  - armv7l (crossbuild)

### CODE REVIEW

**Comment by Piraty** [comment]
ping @the-maldridge

**Comment by pascal-huber** [comment]
Maybe some clarification: @lemmi  expressed some objections on IRC regarding overriding the default `do_build` because there could be issues for cross builds. However, I could not get it to compile with `go mod`/`go install` as in the  go build style. The upstream way is to use the Makefile (which sets some variables and runs `go build`). From what I can see, the `go` build style takes care of the necessary variables (most importantly `GOARCH`).

**Comment by lemmi** [comment]
It works by changing the `go_import_path` to
```sh
go_import_path="github.com/git-lfs/git-lfs/v3"
```
See https://github.com/git-lfs/git-lfs/blob/02ac3de0e19fbfeea7f8ddb079cfe074367a48d3/go.mod#L3

**Comment by pascal-huber** [comment]
> It works by changing the `go_import_path` to
> 
> ```shell
> go_import_path="github.com/git-lfs/git-lfs/v3"
> ```

Oh nice! Changed it, thanks!

File: srcpkgs/git-lfs/template [hunk 1]
```diff
@@ -1,52 +1,24 @@
...
 short_desc="Git extension for versioning large files"
 maintainer="Michael Aldridge <maldridge@voidlinux.org>"
 license="MIT"
 homepage="https://git-lfs.github.com/"
 distfiles="https://github.com/git-lfs/git-lfs/archive/v${version}.tar.gz"
-checksum=2b2e70f1233f7efe9a010771510391a07527ec7c0af721ecf8edabac5d60f62b
+checksum=d5eeb9ee33188d3dd6a391f8a39b96d271f10295129789e5b3a1ac0e9f5114f5
 
 post_build() {
 	make man
```
Comment (lemmi) [inline srcpkgs/git-lfs/template:16]: https://github.com/void-linux/void-packages/actions/runs/3985990165/jobs/6834016287#step:7:2656
```
fatal: not a git repository (or any of the parent directories): .git
...
```
Can be fixed by using
```
VERSION="${version}" make man
```

Comment (lemmi) [reply srcpkgs/git-lfs/template:16]: See https://github.com/git-lfs/git-lfs/blob/02ac3de0e19fbfeea7f8ddb079cfe074367a48d3/Makefile#L8

**Review by lemmi — COMMENTED** [review]
There are still some error that can easily be silenced.

File: srcpkgs/git-lfs/template (hunk 1 above)
Comment (pascal-huber) [reply srcpkgs/git-lfs/template]: good catch - the error was already present in v2 but seemed not to be fatal.

done.

**Comment by pascal-huber** [comment]
> And since we are at it, maybe add a link to the [changelog](https://github.com/void-linux/void-packages/blob/master/Manual.md#optional-variables) as well.

added (raw) changelog.

> There are still some error that can easily be silenced. 

Seems to be error-free with the fixed `make man`.

**Pull request merged**